documents/process/rust_versions.md

```diff
@@ -26,3 +26,116 @@ When the nightly cronjob CI fails, ICU4X developers must fix it and ensure it do
 
 
  [stability-policy]: https://rust-lang.github.io/rfcs/1122-language-semver.html
+
+
+### Pinned Rust nightly policy
+
+ICU4X pins a Rust Nightly in CI for multiple purposes:
+
+ - `diplomat-coverage/src/main.rs`: Running the diplomat-coverage task, which uses unstable Rustdoc JSON output, requiring synchronization between the Rust version and the `rustdoc_types` dependency.
+ - `PINNED_CI_NIGHTLY`: Running various FFI CI jobs that need _some_ Rust nightly to use build-std or other nightly features, without becoming beholden to arbitrary
+ - `LLVM_COMPATIBLE_NIGHTLY`: Usage of `-Clinker-plugin-lto` in size-optimized tests where Rust and Clang need to be using compatible LLVM versions.
+
+
+ ICU4X also runs its entire CI against the latest nightly once a day, reporting errors to Slack. These errors help us catch upstream breakages early, but ICU4X does not have any particular SLO against these being fixed. We guarantee we build on the latest stable as detailed in the first section of this page.
+
+ The first two nightlies here can be updated whenever necessary to whatever nightly works. It not necessary to keep the nightly versions the same, though it is ideal to try.
+
+
+ We have a stricter policy around updating `LLVM_COMPATIBLE_NIGHTLY`. We recommend usage of ICU4X with `-Clinker-plugin-lto` for size-constrained clients who have the option available, so `LLVM_COMPATIBLE_NIGHTLY` must be updated to nightlies that work with "widely available LLVM" versions. This may further constrain our MSRV policy above in case `LLVM_COMPATIBLE_NIGHTLY` cannot be pinned to something that works with an otherwise-allowed MSRV.
+
+An `LLVM_COMPATIBLE_NIGHTLY` is said to "work with" an LLVM version when the size test benchmarks (makefiles in `tutorials/c-tiny/`) run without erroring due to mismatched LLVM metadata. Note that Rust uses LLVM trunk, so it may happen that a Rust nightly is not compatible with _any_ released LLVM version. Some experimentation is often required to find a nightly/LLVM pair that works.
+
+
+"widely available LLVM" here means that:
+
+ - It MUST be available via apt on all currently-supported Ubuntu LTS
+ - It MUST be available via apt on Debian `testing`
+ - It MUST be available via `brew`
+ - It MUST be available on Fedora via `yum`
+ - It MUST be available on the GitHub Actions runners with `ubuntu-latest`
+ - It SHOULD be available on the latest XCode
+ - It SHOULD be available on ICU4X developer machines using nonstandard package management (more on this below)
+ - It SHOULD be available on RHEL and latest Rocky Linux via `yum`
+ 
+
```

Comment:
Suggestion: we should say that the difference between "must" and "should" is that "must" is always required and "should" is something where "the ICU4X developer must make a per-case argument as to why this update is beneficial to the project" (language from above).

Reply:
Yes, though we already have individual policies listed below and they expand on what this means. I'll add.